File: videojuegoDeChinpokomon/src/chimpokomon/GimnasioChimpokomon.java
package Chimpokomon;

import Ataque.*;
import Naturaleza.Naturaleza;
import logger.*;



import static Chimpokomon.ChimpokomonFactory.Zapato;
import static Chimpokomon.ChimpokomonFactory.ZapatoConDosAtaques;
import static Chimpokomon.ChimpokomonFactory.Carnotron;
import static Chimpokomon.ChimpokomonFactory.Gallotronix;

public class GimnasioChimpokomon {

    protected Chimpokomon chinpokomonA;
    protected Chimpokomon chinpokomonB;
    public static Logger logger = Logger.getInstance();

    public GimnasioChimpokomon(Chimpokomon chinpokomonA, Chimpokomon chinpokomonB) {
        this.chinpokomonA = chinpokomonA;
        this.chinpokomonB = chinpokomonB;

    }

    public void peleaChinpokomones(){
        while(!this.chinpokomonA.estaDerrotado() && !this.chinpokomonB.estaDerrotado()){

            chinpokomonA.pelearContra(chinpokomonB);
            logger.info("La vida de "+chinpokomonA.getNombre() + " es: " + chinpokomonA.getNivelDeVida());
            logger.info("La vida de "+chinpokomonB.getNombre() + " es: " + chinpokomonB.getNivelDeVida());
            if(!this.chinpokomonA.estaDerrotado() && !this.chinpokomonB.estaDerrotado()){
                chinpokomonB.pelearContra(chinpokomonA);
            }
        }
        logger.info("La vida de "+chinpokomonA.getNombre() + " es: " + chinpokomonA.getNivelDeVida());
        logger.info("La vida de "+chinpokomonB.getNombre() + " es: " + chinpokomonB.getNivelDeVida());
        logger.info("Tenemos un ganador!");
        if(chinpokomonB.estaDerrotado()){
            logger.info("El ganador es: "+ chinpokomonA.getNombre());
        }
        else{
            logger.info("El ganador es: "+ chinpokomonB.getNombre());
        }
    }

    public static void main(String[] args) {

        // GimnasioChimpokomon gimnasioChimpokomon = new GimnasioChimpokomon(Zapato(Naturaleza.robot), ZapatoConDosAtaques(Naturaleza.animal));
        GimnasioChimpokomon gimnasioChimpokomon = new GimnasioChimpokomon(Carnotron(Naturaleza.robot), Gallotronix(Naturaleza.animal));

        gimnasioChimpokomon.peleaChinpokomones();










    }
}

File: ChimpokomonPython/ModuloSRC/GimnasioChimpokomon.py
from Chimpokomons.ChimpokomonFactory import ChimpokomonFactory
from Logger.Level import Level
from Logger.Logger import Logger
from Naturaleza.Naturaleza import Naturalezas

class GimnasioChimpokomon():

    def __init__(self, chimpokomonAtacante, chimpokomonEnemigo):
        self.chimpokomonAtacante = chimpokomonAtacante
        self.chimpokomonEnemigo = chimpokomonEnemigo


    Logger.instance(Level.INFO)

    def peleaChimpokomones(self):
        while not self.chimpokomonAtacante.estaDerrotado() and not self.chimpokomonEnemigo.estaDerrotado():
            self.chimpokomonAtacante.pelearContra(self.chimpokomonEnemigo)
            Logger.instance().info('La vida de ' + str(self.chimpokomonAtacante.nombre) + ' es ' + str(self.chimpokomonAtacante.nivelDeVida))
            self.chimpokomonEnemigo.pelearContra(self.chimpokomonAtacante)
            Logger.instance().info('La vida de ' + str(self.chimpokomonEnemigo.nombre) + ' es ' + str(self.chimpokomonEnemigo.nivelDeVida))
        if self.chimpokomonEnemigo.estaDerrotado():
            Logger.instance().info(str(self.chimpokomonAtacante.nombre) + ' Es el ganador')
        else:
            Logger.instance().info(str(self.chimpokomonEnemigo.nombre) + ' Es el ganador')

Logger.instance(Level.INFO)
batallar = GimnasioChimpokomon(ChimpokomonFactory.Zapato(Naturalezas.COSA), ChimpokomonFactory.ZapatoConDosAtaques(Naturalezas.COSA))
batallar.peleaChimpokomones()
        



# if __name__ == "__main__":
#         print("Bienvenido al gimnasio de Chimpokomon")
      

#         gimnasio = GimnasioChimpokomon(ChimpokomonFactory.Zapato(Naturalezas.COSA), ChimpokomonFactory.ZapatoConDosAtaques(Naturalezas.COSA))

#         gimnasio.peleaChimpokomones()

#         print(gimnasio.log1._info)
